lua/efmls-configs/linters/slither.lua

```diff
@@ -0,0 +1,44 @@
+local fs = require('efmls-configs.fs')
+
+local linter = 'slither'
+local bin = fs.executable(linter)
+local args = "--sarif - ${INPUT}"
+-- `solc` needs to be manually configured, see:
+-- https://github.com/foundry-rs/foundry/issues/5429
+-- https://github.com/crytic/slither/issues/2054
+-- Also, `remappings.txt` needs to generated, e.g. `forge remappings > remappings.txt`
+local command = string.format([[
+cmd="%s %s"
+(if [ -f remappings.txt ]; then
+    $cmd --solc-remaps "$(tr '\n' ' ' < remappings.txt | xargs)"
+else
+    $cmd
+fi) \
+| jq -r '.runs[].results[]
+| "\(.locations[].physicalLocation.artifactLocation.uri):\(
+    .locations[].physicalLocation.region.startLine)-\(
+    .locations[].physicalLocation.region.endLine): \(
+    .message.text | gsub("\\s+";" ";"g"))"'
+]], bin, args)
```

Comment:
Is there any other way to not do this? I see that `jq` is required for this and seems like its written in `bash` so lots of different tooling would be required and installed beforehand.

Reply:
That implementation was POSIX-compliant other than the call to `jq`, but I just updated it to be fully POSIX-compliant so only `slither` is necessary.

The implementation is a lot more verbose/nasty since POSIX is missing a lot of features, but the upside is that it's extremely fast (`jq` is kind of slow) and it gets more information (`jq` version didn't have start and end column or severity)

This new implementation is the probably the best that can be done without upstreaming a change and I did file an issue with this tool's maintainers, but IDE support doesn't seem to be their priority right now.